### DISCUSSION TITLE
REST API: `/notifications` endpoint doesn't include certain notifications

### DISCUSSION THREAD

**Discussion by bastimeyer** [community/community#46174] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

When requesting notifications of an authenticated user on the [REST API's `/notifications` endpoint](https://docs.github.com/en/rest/activity/notifications?apiVersion=2022-11-28#list-notifications-for-the-authenticated-user) (version 2022-11-28), I'm expecting all notifications to be listed there which are also shown on the notifcations page on the GitHub website. This is what the API docs say, at least.

For example, I'm using a [browser extension](https://github.com/sindresorhus/notifier-for-github) for showing the number of unread notifications of my GH account and for triggering desktop notifications when a new unread notification appears. The extension requires a PAT with the notifications permissions ("classic" token) and it is polling the API endpoint while caching notification IDs in-between API requests.

However, as it turns out, some notification types are not included in the API responses (tested via `curl`, so the browser extension is not at fault here). This means that notifications for private organisation/team discussions can't be shown by the browser extension, and it requires me to stay alerted on the website itself when a new reply is being made in one of the private team discussions. This unfortunately defeats the purpose of the browser extension or other applications which rely on the correctness of API endpoint's data.

The `/notifications` REST API endpoint should include notifications of all kinds.

How to reproduce:
1. Create a private team discussion in one of your orgas
2. Have someone reply and make sure to have the event marked as unread on the notifications page on the GH website
3. Request the REST API endpoint with your PAT and see no notification for the private team discussion being included there

Thanks!

**Comment by imidg1825** [comment]
This is a test answer for the Galaxy Brain achievement.
